flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/ProjectionTranslator.scala

```diff
@@ -227,18 +227,23 @@ object ProjectionTranslator {
     * @param exprs a list of expressions to extract
     * @return a list of field references extracted from the given expressions
     */
-  def extractFieldReferences(exprs: Seq[Expression]): Seq[NamedExpression] = {
-    exprs.foldLeft(Set[NamedExpression]()) {
+  def extractFieldReferences(exprs: Seq[Expression]): List[NamedExpression] = {
+    exprs.foldLeft(List[NamedExpression]()) {
       (fieldReferences, expr) => identifyFieldReferences(expr, fieldReferences)
-    }.toSeq
+    }
   }
 
   private def identifyFieldReferences(
       expr: Expression,
-      fieldReferences: Set[NamedExpression]): Set[NamedExpression] = expr match {
+      fieldReferences: List[NamedExpression]): List[NamedExpression] = expr match {
```

Comment:
We can also use a `LinkedHashSet` which preserves the order in which elements are inserted.